src/components/input/Button/Button.styles.js

```diff
@@ -1,65 +1,85 @@
 import styled, { css } from 'styled-components';
 
 import { buttonColors } from '../../../base/color/colorVariables';
-import fontSizes from '../../../base/typography/fontSizes';
+import { interfaceTypography } from '../../../base/typography/typographyVariables';
+import spacerSizes from '../../../base/spacing';
 
-export const buttonModes = {
+
+// We're going to try to colour the buttons based on state or intent
+export const setButtonColour = (intent) => {
+  console.log(intent);
+  //const currentButtonColour = buttonColors[intent];
+  console.log(currentButtonColour);
+  return currentButtonColour;
+};
+
+export const buttonProminences = (intent) => {
   primary: css`
-    background-color: ${buttonColors.buttonPrimaryBackground};
-    border: 1px solid ${buttonColors.buttonPrimaryBorder};
-    color: ${buttonColors.buttonPrimaryText};
+    background-color: ${setButtonColour(intent)};
+    border: none;
+    border-radius: 10px;
+    color: ${buttonColors.primaryText};
 
     &:hover {
-      background-color: ${buttonColors.buttonPrimaryBackgroundHover};
+      background-color: ${buttonColors.hover};
     }
   `,
-  secondary: css`
-    background-color: ${buttonColors.buttonSecondaryBackground};
-    border: 1px solid ${buttonColors.buttonSecondaryBorder};
-    color: ${buttonColors.buttonSecondaryText};
+  default: css`
+    background: none;
+    border: 2px solid ${buttonColors.default};
+    border-radius: 10px;
+    color: ${buttonColors.default};
 
     &:hover {
-      background-color: ${buttonColors.buttonSecondaryBackgroundHover};
+      border-color: ${buttonColors.hover};
+      color: ${buttonColors.hover};
     }
   `,
-  danger: css`
-    background-color: ${buttonColors.buttonDangerBackground};
-    border: 1px solid ${buttonColors.buttonDangerBorder};
-    color: ${buttonColors.buttonDangerText};
+  minimal: css`
+    background: none;
+    border: none;
+    border-radius: 10px;
+    color: ${buttonColors.default};
 
     &:hover {
-      background-color: ${buttonColors.buttonDangerBackgroundHover};
+      color: ${buttonColors.hover};
     }
   `,
-  success: css`
-    background-color: ${buttonColors.buttonSuccessBackground};
-    border: 1px solid ${buttonColors.buttonSuccessBorder};
-    color: ${buttonColors.buttonSuccessText};
+};
 
-    &:hover {
-      background-color: ${buttonColors.buttonSuccessBackgroundHover};
-    }
+// success, warning, danger, none
+export const buttonIntents = {
+  success: css`
+    
+    }, 
```

Comment:
This doesn't seem finished...

Reply:
Yes! There are still lots of pieces that aren't finished—I need to get the basics sorted out first, and then I can work on the things that build on it.